### PR TITLE
 kubeadm: remove iptables/ip/tc/ethtool binary preflight check

### LINKED ISSUE #3132: kubeadm preflight support for nftables kube-proxy / hosts with no iptables
The kubeadm preflight checks [currently consider the `iptables` binary to be mandatory](https://github.com/kubernetes/kubernetes/blob/v1.32.0-rc.2/cmd/kubeadm/app/preflight/checks_linux.go#L87). This is not correct if you are running kube-proxy in nftables mode, and while it mostly doesn't cause problems now (since pretty much everyone has `iptables` installed anyway), it will eventually be a problem on RHEL10, where there won't be any `iptables` binaries.

It appears that kubeadm has a standard way of overriding the kube-proxy configuration, so it should be able to parse the provided config to see if it overrides the `mode` to be `nftables`, and in that case, it should require the `nft` binary and _not_ require the `iptables` binary.

(For extra credit: `ipvs` mode requires the `iptables` and `ipset` binaries. It doesn't actually require any IPVS-specific binaries because it uses the netlink API to program IPVS.)

/sig cluster-lifecycle
/area kubeadm

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

fixes https://github.com/kubernetes/kubeadm/issues/3132

#### Special notes for your reviewer:

Should this be updated as well?
https://github.com/kubernetes/kubernetes/blob/ad0f6e29ddc6f2c6d96593f02031dfc586e4fc72/cmd/kubeadm/app/cmd/reset.go#L51-L52
- [ ] todo: fix kubelet log which is an error message per minute.

#### Does this PR introduce a user-facing change?
```release-note
kubeadm: removed preflight check for `ip`, `iptables`, `ethtool` and `tc` on Linux nodes. kubelet and kube-proxy will continue to report `iptables` errors if its usage is required. The tools `ip`, `ethtool` and `tc` had legacy usage in the kubelet but are no longer required.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sftim** [comment]
Changelog suggestion
```diff
-kubeadm: remove iptables preflight check
+kubeadm: removed preflight check for `iptables` on Linux nodes
```

**Comment by SataQiu** [comment]
Did we verify that the e2e tests passed completely without iptables binary?

**Comment by pacoxu** [comment]
> Did we verify that the e2e tests passed completely without iptables binary?

/hold
will do

**Comment by pacoxu** [comment]
https://github.com/neolit123/kubeadm-test/actions/runs/12310425050
passed.

**Review by neolit123 — COMMENTED** [review]
maybe we should remove all the tools that we discussed on the issue ticket.
ip,iptables,ethtool,tc
https://github.com/kubernetes/kubeadm/issues/3132#issuecomment-2540422690

nsenter can be removed later once @carlory 's PR merges.

**Comment by pacoxu** [comment]
> maybe we should remove all the tools that we discussed on the issue ticket. ip,iptables,ethtool,tc [kubernetes/kubeadm#3132 (comment)](https://github.com/kubernetes/kubeadm/issues/3132#issuecomment-2540422690)

Updated.

> nsenter can be removed later once @carlory 's PR merges.

May Baofa/Kiki remove the check in that PR https://github.com/kubernetes/kubernetes/pull/122016?

**Comment by neolit123** [comment]
/release-note-edit

```release-note
kubeadm: removed preflight check for `ip`, `iptables`, `ethtool` and `tc` on Linux nodes. kubelet and kube-proxy will continue to report `iptables` errors if its usage is required. The tools `ip`, `ethtool` and `tc` had legacy usage in the kubelet but are no longer required.
```

**Comment by neolit123** [comment]
> 
> > nsenter can be removed later once @carlory 's PR merges.
> 
> May Baofa/Kiki remove the check in that PR #122016?

yes

File: cmd/kubeadm/app/preflight/checks_linux.go [hunk 1]
```diff
@@ -83,12 +83,8 @@ func addExecChecks(checks []Checker, execer utilsexec.Interface, k8sVersion stri
 	}
 
 	checks = append(checks,
-		InPathCheck{executable: "ip", mandatory: true, exec: execer},
-		InPathCheck{executable: "iptables", mandatory: true, exec: execer},
 		InPathCheck{executable: "mount", mandatory: true, exec: execer},
 		InPathCheck{executable: "nsenter", mandatory: true, exec: execer},
-		InPathCheck{executable: "ethtool", mandatory: false, exec: execer},
-		InPathCheck{executable: "tc", mandatory: false, exec: execer},
 		InPathCheck{executable: "touch", mandatory: false, exec: execer})
```
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks_linux.go:88]: touch is likely not required either, but let's keep it in a separate PR.

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm

**Comment by neolit123** [comment]
> xref https://github.com/kubernetes/kubeadm/issues/3132

this can be `fixes ...`
